### ISSUE TITLE
Fuzzing results: segmentation fault upon erroneous CUE file input

### ISSUE THREAD

**Issue by marisgg** [flacon/flacon#166]
Hello,

The following function produces a segmentation fault if the index argument is erroneous with respect to the `mFileTags` variable. We found this by fuzzing the input CUE files supplied to Flacon in the silent mode `-s` for a Software Security course at Radboud University.

```cpp
QStringList AudioFileMatcher::audioFiles(int index) const
{
    QString tag = mFileTags[index];
    return audioFiles(tag);
}
```

This is the trace when Flacon is compiled with ASan:

```
Start flacon  8.1.0  + git  master   01cfa5f5cdbccee9a2761990667a28cff38baf46
AddressSanitizer:DEADLYSIGNAL
=================================================================
==765==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x55addc69497c bp 0x7ffea46135b0 sp 0x7ffea46134d0 T0)
==765==The signal is caused by a READ memory access.
==765==Hint: address points to the zero page.
    #0 0x55addc69497b in AudioFileMatcher::audioFiles(int) const (/usr/local/bin/flacon+0x1b697b)
    #1 0x55addc622f30 in Disc::searchAudioFiles(bool) (/usr/local/bin/flacon+0x144f30)
    #2 0x55addc649a60 in Project::addCueFile(QString const&) (/usr/local/bin/flacon+0x16ba60)
    #3 0x55addc5fd0c0 in runConsole(int, char**, QStringList const&) (/usr/local/bin/flacon+0x11f0c0)
    #4 0x55addc59cd0f in main (/usr/local/bin/flacon+0xbed0f)
    #5 0x7f901afe60b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #6 0x55addc59e16d in _start (/usr/local/bin/flacon+0xc016d)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/usr/local/bin/flacon+0x1b697b) in AudioFileMatcher::audioFiles(int) const
==765==ABORTING
```

This function is found in the `audiofilematcher.cpp` source file.

You can reproduce this with zzuf easily. To reproduce with zzuf, run `zzuf -s 20 flacon -s orig.cue`. 

The `orig.cue` file can be found on our Github repository: https://github.com/marisgg/fuzzy-software/blob/main/orig.cue.

You can also reproduce this error by supplying the `zufseg.cue` file to Flacon from the command line. It can also be found on our repository: https://github.com/marisgg/fuzzy-software/blob/main/segfaults/zufseg.cue.

One might argue that the significance of this error is small, since Flacon expects a properly formatted CUE file as input. However, segmentation faults can be dangerous; we thus wanted to share this finding. The minimal example to reproduce this error is to replace the space after the FILE tag with any other string of characters.

**Comment by SokoloffA** [comment]
Thank you so much! 
I'm analyzing the problem. And, of course, I will fix it

**Comment by SokoloffA** [comment]
Closed in release 8.2.0
